posts/a-beginners-guide-to-influxdb.md

```diff
@@ -0,0 +1,109 @@
+
+# Line Protocol
+The line protocol is the text-based format that uses below syntax to write points to InfluxDB.
+
+``` <measurement>[,<tag_key>=<tag_value>[,<tag_key>=<tag_value>]] <field_key>=<field_value>[,<field_key>=<field_value>] [<timestamp>] ```
+
+The above syntax is used to insert a single point to an InfluxDB database. Since InfluxDB is a time-series database, the **point** here is analogous to a point on a graph that is representing a time-series data. And the collection of points is called a **series**. Now, let’s look at all the four components of the point.
+
+The first component, **Measurement** for an InfluxDB is similar to a Table for a SQL database. Measurement holds all the tags, fields and the time values and it makes sense to name the measurement such that it describes the data contained in it.
+
+The two components, **tag set** and **field set**, in the line protocol are sets, meaning a pair of key and value. **Tag Key** and **Tag Value** are both stored in InfluxDB as strings, while only **Field Key** is string but the **Field Value** can contain strings, integers, floats and booleans too.  The field value is the actual data that we want to save to a database and it is always accompanied by a timestamp, which makes sense since we are inserting a record to a time-series database. As you have guessed by now, the field value, which is the actual data or the field set in general as well as a timestamp are essential components without which we can not save a point to the database. Although tag set is an optional component, it is a good practice to use tag set, since tag set is indexed and would be useful in faster querying of data, as opposed to the field set, which is not indexed.
+
+The last and the most important component of the line protocol is **time**, which is a **timestamp** in the RFC3339 UTC format. The timestamp precision can be in second, millisecond, microsecond or nanosecond. 
+
+
+# Writing Data to InfluxDB
+Data can be written to InfluxDB using command line interface using InfluxDB API or using client libraries. A simple curl, command line tool can be used to create and write to a database. InfluxDB provides client libraries for most of the famous languages like Java, PHP, Python, C++, Go etc. I have used Python influxdb-client and below is an example of how to write the data using python influxdb-client.
+
+ - Start with importing the required modules in your program
+
+    ``` python
+    from influxdb_client import InfluxDBClient, Point, WritePrecision
+    from influxdb_client.client.write_api import SYNCHRONOUS 
+    ```
+
+ - Then create an InfluxDB client (mention your url, token and org as below) and a writer object
```

Comment:
- Then create an InfluxDB client (mention your URL, token, and org, as below) and a writer object